test/testdrive/mz-global-id.td

```diff
@@ -0,0 +1,61 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+> CREATE DATABASE d;
+> CREATE SCHEMA s;
+> CREATE SCHEMA d.s;
+
+> CREATE SOURCE counter
+  FROM LOAD GENERATOR COUNTER (TICK INTERVAL '1s');
+
+> CREATE SOURCE s.counter
+  FROM LOAD GENERATOR COUNTER (TICK INTERVAL '1s');
+
+> CREATE SOURCE d.s.counter
+  FROM LOAD GENERATOR COUNTER (TICK INTERVAL '1s');
+
+# GlobalIDs are not stable across runs, so best we can do is ensure they roundtrip.
+
+> SELECT
+    mz_internal.mz_global_id_to_name(
+        mz_internal.mz_name_to_global_id('counter')
+    );
+materialize.public.counter
+
+> SELECT
+    mz_internal.mz_global_id_to_name(
+        mz_internal.mz_name_to_global_id('s', 'counter')
+    );;
+materialize.s.counter
+
+> SELECT
+    mz_internal.mz_global_id_to_name(
+        mz_internal.mz_name_to_global_id(
+            'd', ARRAY['s'], 'counter'
+        )
+    );
+d.s.counter
+
+! SELECT mz_internal.mz_name_to_global_id(ARRAY['public', 's'], 'counter');
+contains:Evaluation error: more than one record produced in subquery
```

Comment:
I have trouble understanding this query and the error it throws

Reply:
oh the subquery in question refers to the subquery in the implementation of the function.

Reply:
Improved this by making it a table function; ty for noting that this was odd/unclear.